figures/pipeline/course_daily_metrics.py

```diff
@@ -145,25 +146,21 @@ def get_days_to_complete(course_id, date_for):
     When we have to support scale, we can look into optimization
     techinques.
     """
-    certificates = GeneratedCertificate.objects.filter(
+    grades = PersistentCourseGrade.objects.filter(
         course_id=as_course_key(course_id),
-        created_date__lte=as_datetime(date_for))
+        passed_timestamp__isnull=False,
+        passed_timestamp__lte=as_datetime(date_for),
+    ).values('user_id', 'passed_timestamp')
```

Comment:
@zubair-arbi @ziafazal I discussed this with @iam-mhaseeb as well. for the time being, he recommended getting only required values instead of whole objects to reduce memory load.